Specifications/QIR/Reserved-Operations.md

```diff
@@ -0,0 +1,41 @@
+| Z | [`__quantum__qis__z__body`](https://github.com/microsoft/qsharp-runtime/blob/5a379fd1c72ea19cb50d6acc501c6896894e96ac/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll#L276) | The [Z operation](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.z) (Parity Gate) | [Single-qubit operations](https://docs.microsoft.com/en-us/azure/quantum/concepts-the-qubit#single-qubit-operations), [Pauli Operators](https://docs.microsoft.com/en-us/azure/quantum/user-guide/libraries/standard/prelude#pauli-operators), ![Z = T^4 = S^2 = HXH](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+Z+%3D+T%5E4+%3D+S%5E2+%3D+HXH) |
+| X | [`__quantum__qis__x__body`](https://github.com/microsoft/qsharp-runtime/blob/5a379fd1c72ea19cb50d6acc501c6896894e96ac/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll#L250) | The [X operation](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.x) (NOT Gate) | [Single-qubit operations](https://docs.microsoft.com/en-us/azure/quantum/concepts-the-qubit#single-qubit-operations), [Pauli Operators](https://docs.microsoft.com/en-us/azure/quantum/user-guide/libraries/standard/prelude#pauli-operators), ![X = HT^4H = HZH](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+X+%3D+HT%5E4H+%3D+HZH) |
+| Y | [`__quantum__qis__y__body`](https://github.com/microsoft/qsharp-runtime/blob/5a379fd1c72ea19cb50d6acc501c6896894e96ac/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll#L263) | The [Y operation](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.y) | [Single-qubit operations](https://docs.microsoft.com/en-us/azure/quantum/concepts-the-qubit#single-qubit-operations), [Pauli Operators](https://docs.microsoft.com/en-us/azure/quantum/user-guide/libraries/standard/prelude#pauli-operators), ![Y = T^2HT^4HT^6 = SXZS = SXS^\dagger](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+Y+%3D+T%5E2HT%5E4HT%5E6+%3D+SXZS+%3D+SXS%5E%5Cdagger) |
+| Rx | [`__quantum__qis__r__body`](https://github.com/microsoft/qsharp-runtime/blob/5a379fd1c72ea19cb50d6acc501c6896894e96ac/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll#L148), see [R](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.r) | The [Rx operation](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.rx) | [Single-qubit operations](https://docs.microsoft.com/en-us/azure/quantum/concepts-the-qubit#single-qubit-operations), [Rotations](https://docs.microsoft.com/en-us/azure/quantum/user-guide/libraries/standard/prelude#rotations) |
```

Comment:
These three should have different QIR implementation names: `__quantum__qis__rx__body`, `__quantum__qis__ry__body`, and `__quantum__qis__rz__body`. The descriptions should be stand-alone; for instance, Rx is the X rotation gate.

Reply:
I assume that with "These three" you mean "Z, X, Y" (the selection shows four: "Z, X, Y, Rx").

Then, I'm not strong in quantum computing, but my understanding is that [X](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.x) and [Rx](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.rx) are different operations (maybe X is a particular case of Rx, but I'm not sure). That is why it would be wrong thing if X operation was implemented under the name `__quantum__qis__rx__body`.

Please provide more details.

Reply:
Sorry, the selection wasn't quite right -- the three are Rx, Ry, and Rz, which should be implemented under 3 different names.